### DISCUSSION TITLE
Implement match "priority" feature for disambiguation

### DISCUSSION THREAD

**Discussion by Masacroso** [espanso/espanso#1233] [Feature requests and Ideas]
I was using espanso these days, testing my old dictionary to write LaTeX faster, and the regex is insanely useful and very fast (at least for very short sequences). However I think that something in how regex handles the matches could be improved, by now when you have two regex that match a string then a popup show that these two regex matched so you can choose some of the two to be applied, but it will be (extremely) useful an option to make the selection of the greediest match automatic (this is similar how regular matches works in espanso). I will explain with an example, let

      - regex: (?P<name1>\w)j
        replace: {{name1}}
      - regex: (?P<name1>\w)(?P<name2>\w)j
        replace: {{name1}} {{name2}}

Then if I write "xyj" both regex match it, and the popup shows. Then, what Im saying, is that it will be useful that the greediest match will be automatically used, that is, that the second regex will be applied without asking what of both must be applied. Indeed regular matches in espanso works in this greedy fashion, by example if we have defined the replacements

      - match: xy
        replace: "Hello, you had typed xy."
      - match: xxy
        replace: "You had typed xxy."

and type "xxy" then the second match occur.

**Comment by federico-terzi** [comment]
Hey @Masacroso,

Thank you for the feedback! This feature is pretty tricky to implement with the current architecture, as the "match disambiguation" (aka the selection box), is built at a different level compared to the regex triggers, and have no notion of it.
I'm wondering if we can add an additional flag like `priority: X` that can be applied to a match and, if no other match has the same priority, then it select the one with the highest priority without showing the selection box, something like:

```
  - regex: (?P<name1>\w)j
    replace: {{name1}}
  - regex: (?P<name1>\w)(?P<name2>\w)j
    priority: 100
    replace: {{name1}} {{name2}}
```

which would always expand to the second one if both matched. This would have no knowledge of regex greediness though

Would this solve your use-case? 

Cheers :)

**Comment by Masacroso** [comment]
> Would this solve your use-case?
> 
> Cheers :)

Yes, absolutely. If you can implement it this will be awesome. Thank you very much Federico for your generosity.
